src/components/modals/SaveCopy.js

```diff
@@ -0,0 +1,123 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import Modal from "../common/Modal";
+import axios from 'axios';
+import constants, { VERSION } from '../../constants';
+import { getAuthHeader } from "../../utils/Auth";
+
+const SaveCopy = () => {
+    const dispatch = useDispatch();
+    const map = useSelector(state => state.map);
+    const drawings = useSelector(state => state.drawings);
+    const mapLayers = useSelector(state => state.mapLayers);
+    const markers = useSelector(state => state.markers);
+    const { activeDataGroups } = useSelector(state => state.dataGroups);
+    const { isSnapshot } = useSelector(state => state.mapMeta);
+
+    if (!map.name)
+        return <Modal id="saveCopy">
+            <div className="modal-title">Save a copy</div>
+            <div className="modal-content">
+                <div>Please add a title to save the map first</div>
+            </div>
+        </Modal>
+
+    const saveMap = () => {
+        const activeDataGroupsInfo = activeDataGroups.map(group => ({
+            iddata_groups: group.iddata_groups,
+            title: group.title,
+            userGroupId: group.userGroupId
+        }));
+        const saveData = {
+            map: {
+                ...map,
+                gettingLocation: false,
+                name: map.name,
+                currentLocation: null,
+                searchMarker: null,
+            },
+            drawings: drawings,
+            markers: markers,
+            mapLayers: {
+                landDataLayers: mapLayers.landDataLayers,
+                myDataLayers: activeDataGroupsInfo
+            },
+            version: VERSION,
+            name: map.name,
+
+        };
+
+        console.log(saveData)
+
+        const body = {
+            "eid": null,
+            "name": map.name,
+            "data": JSON.stringify(saveData),
+            "isSnapshot": isSnapshot
+        }
+        axios.post(`${constants.ROOT_URL}/api/user/map/save/`, body, getAuthHeader())
```

Comment:
I think reusing the standard save API is mostly okay because eid=null, so a new map is created. But I think some backend changes might be needed, to make sure that the objects are not saved with duplicate UUIDs in the database.

Also, maybe it's worth moving this business logic out of the component, into a thunk in the /actions folder? It's also very similar to the code in Save.js, so might get rid of some duplication

Reply:
I agree on putting this in a thunk. Going to deploy the branch tonight then do it tomorrow then merge with development

Reply:
Done in PR #183 since I needed to do it anyway for the below autosave bugfix